core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableSortedAggregate.java

```diff
@@ -90,6 +101,133 @@ public EnumerableSortedAggregate(
   }
 
   public Result implement(EnumerableRelImplementor implementor, Prefer pref) {
-    throw Util.needToImplement("EnumerableSortedAggregate");
+    if (getGroupType() != Group.SIMPLE
```

Comment:
Would it be possible to extract these conditions as an auxiliary method `isSupported` ?
By doing this, we could make EnumerableSortedAggregateRule to check it, and it case of not supported, do not generate the EnumerableSortedAggregate (otherwise the rule would generate an operator that will fail when trying to implement).

Reply:
That'a good point. In fact, I think I should follow https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableSortedAggregateRule.java#L41